lib/repl.js

```diff
@@ -359,7 +354,11 @@ function REPLServer(prompt,
         if (self.breakEvalOnSigint) {
           const interrupt = new Promise((resolve, reject) => {
             sigintListener = () => {
-              reject(new ERR_SCRIPT_EXECUTION_INTERRUPTED());
+              const tmp = Error.stackTraceLimit;
+              Error.stackTraceLimit = 0;
+              const err = new ERR_SCRIPT_EXECUTION_INTERRUPTED();
+              Error.stackTraceLimit = tmp;
+              reject(err);
```

Comment:
Since `.stack` creation is deferred until the `.stack` is accessed does toggling the `Error.stackTraceLimit` have the effect wanted here _(making an empty stack)_? It looks like `stackTraceLimit` is reset before the lazy `.stack` is generated.

Reply:
The stack size is set on error creation time. Otherwise the test would fail as well.

Reply:
Ah, good to know!